pkg/registry/apis/provisioning/lint/dashboard.go

```diff
@@ -0,0 +1,67 @@
+package lint
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/grafana/dashboard-linter/lint"
+)
+
+type DashboardLinter struct {
+	rules lint.RuleSet
+}
+
+// FIXME: what would be a good place to put all the schema validation?
+type specData struct {
+	Spec json.RawMessage `json:"spec"`
+}
+
+func NewDashboardLinter() *DashboardLinter {
+	// TODO: read rules from configuration and pass to the lint function
+	return &DashboardLinter{rules: lint.NewRuleSet()}
+}
+
+func (l *DashboardLinter) Lint(ctx context.Context, fileData []byte) ([]Issue, error) {
+	var data specData
+	if err := json.Unmarshal(fileData, &data); err != nil {
+		return nil, fmt.Errorf("unmarshal file data into spec: %w", err)
+	}
+
+	dashboard, err := lint.NewDashboard(data.Spec)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse dashboard with linter: %v", err)
+	}
+
+	results, err := l.rules.Lint([]lint.Dashboard{dashboard})
+	if err != nil {
+		return nil, fmt.Errorf("failed to lint dashboard: %v", err)
+	}
+
+	byRule := results.ByRule()
+	rules := make([]string, 0, len(byRule))
+	for r := range byRule {
+		rules = append(rules, r)
+	}
+	sort.Strings(rules)
+
+	issues := make([]Issue, 0)
```

Comment:
`var issues []Issue`? or maybe set the list cap to like 100 at least? Issue isn't all too big of a struct